src/Step.cc

```diff
@@ -154,10 +155,13 @@ bool Step::execute(Context& context, CommChannel* comm, StepIndex index)
 
     if (std::holds_alternative<std::string>(result_or_error))
     {
-        const auto& msg = std::get<std::string>(result_or_error);
-        send_message(comm, Message::Type::step_stopped_with_error, msg, Clock::now(),
-                     index);
-        throw ErrorAtIndex(msg, index);
+        const std::string& raw_msg = std::get<std::string>(result_or_error);
+        std::string laundered_msg = raw_msg;
+        remove_abort_markers_from_error_message(laundered_msg);
```

Comment:
:sob: 
And here you see why `remove_abort_markers_...()` should NOT have an IN/OUT parameter, if you need to construct a variable first to stuff that in? This looks like `C`, except for not using pointers.

```c++
auto [laundered_msg, abort_reason] = remove_abort_markers_from_error_message(raw_msg);
```
would look so much nicer.
This is C++17, we have structured bindings :tada:

Well, and `s/laundered_msg/plain_msg/`.

Reply:
Fair enough. I should have known that I would not get away with an in/out parameter. :D